src/net.cpp

```diff
@@ -1153,310 +1153,407 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
+    SOCKET hSocketMax = 0;
+
+    std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
+    GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
+
+    if (recv_select_set.empty() && send_select_set.empty() && error_select_set.empty()) {
+        interruptNet.sleep_for(std::chrono::milliseconds(50));
+        return;
+    }
+
+    for (SOCKET hSocket : recv_select_set) {
+        FD_SET(hSocket, &fdsetRecv);
+        hSocketMax = std::max(hSocketMax, hSocket);
+    }
+
+    for (SOCKET hSocket : send_select_set) {
+        FD_SET(hSocket, &fdsetSend);
+        hSocketMax = std::max(hSocketMax, hSocket);
+    }
+
+    for (SOCKET hSocket : error_select_set) {
+        FD_SET(hSocket, &fdsetError);
+        hSocketMax = std::max(hSocketMax, hSocket);
+    }
+
+    int nSelect = select(hSocketMax + 1, &fdsetRecv, &fdsetSend, &fdsetError, &timeout);
+
+    if (nSelect == SOCKET_ERROR)
+    {
+        int nErr = WSAGetLastError();
+        LogPrintf("socket select error %s\n", NetworkErrorString(nErr));
+        for (unsigned int i = 0; i <= hSocketMax; i++)
+            FD_SET(i, &fdsetRecv);
+        FD_ZERO(&fdsetSend);
+        FD_ZERO(&fdsetError);
+        if (!interruptNet.sleep_for(std::chrono::milliseconds(timeout.tv_usec/1000)))
             return;
+    }
 
-        if (nSelect == SOCKET_ERROR)
-        {
-            if (have_fds)
-            {
-                int nErr = WSAGetLastError();
-                LogPrintf("socket select error %s\n", NetworkErrorString(nErr));
-                for (unsigned int i = 0; i <= hSocketMax; i++)
-                    FD_SET(i, &fdsetRecv);
-            }
-            FD_ZERO(&fdsetSend);
-            FD_ZERO(&fdsetError);
-            if (!interruptNet.sleep_for(std::chrono::milliseconds(timeout.tv_usec/1000)))
-                return;
-        }
+    for (SOCKET hSocket : recv_select_set)
+        if(FD_ISSET(hSocket, &fdsetRecv))
+            recv_set.insert(hSocket);
 
-        //
-        // Accept new connections
-        //
-        for (const ListenSocket& hListenSocket : vhListenSocket)
+    for (SOCKET hSocket : send_select_set)
+        if(FD_ISSET(hSocket, &fdsetSend))
+            send_set.insert(hSocket);
+
+    for (SOCKET hSocket : error_select_set)
+        if(FD_ISSET(hSocket, &fdsetError))
+            error_set.insert(hSocket);
+}
+#else
+void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_set, std::set<SOCKET> &error_set)
+{
+    std::set<SOCKET> recv_select_set, send_select_set, error_select_set;
+    GenerateSelectSet(recv_select_set, send_select_set, error_select_set);
+
+    if (recv_select_set.empty() && send_select_set.empty() and error_select_set.empty()) {
+        interruptNet.sleep_for(std::chrono::milliseconds(50));
+        return;
+    }
+
+    if (interruptNet)
+        return;
+
+    std::vector<struct pollfd> pollfds;
+    struct pollfd pollfd;
+    memset(&pollfd, 0, sizeof(struct pollfd));
```

Comment:
`struct pollfd pollfd = {}` instead of `memset`?

Reply:
done